rtengine/histmatching.cc

```diff
@@ -185,50 +202,48 @@ void RawImageSource::getAutoMatchedToneCurve(std::vector<double> &outCurve)
     
     std::unique_ptr<IImage8> target;
     {
-        int tw = source->getWidth(), th = source->getHeight();
-        float thumb_ratio = float(std::max(tw, th)) / float(std::min(tw, th));
-        float target_ratio = float(std::max(fw, fh)) / float(std::min(fw, fh));
+        RawMetaDataLocation rml;
+        eSensorType sensor_type;
+        double scale;
+        int w = fw / skip, h = fh / skip;
+        std::unique_ptr<Thumbnail> thumb(Thumbnail::loadFromRaw(getFileName(), rml, sensor_type, w, h, 1, false, false));
+        target.reset(thumb->processImage(neutral, sensor_type, fh / skip, TI_Nearest, getMetaData(), scale, false));
+
+        int sw = source->getWidth(), sh = source->getHeight();
+        int tw = target->getWidth(), th = target->getHeight();
+        float thumb_ratio = float(std::max(sw, sh)) / float(std::min(sw, sh));
+        float target_ratio = float(std::max(tw, th)) / float(std::min(tw, th));
         int cx = 0, cy = 0;
         if (std::abs(thumb_ratio - target_ratio) > 0.01) {
             if (thumb_ratio > target_ratio) {
                 // crop the height
-                int ch = fh - (fw * float(th) / float(tw));
+                int ch = th - (tw * float(sh) / float(sw));
                 cy += ch / 2;
-                fh -= ch;
+                th -= ch;
             } else {
                 // crop the width
-                int cw = fw - (fh * float(tw) / float(th));
+                int cw = tw - (th * float(sw) / float(sh));
                 cx += cw / 2;
-                fw -= cw;
+                tw -= cw;
             }
             if (settings->verbose) {
-                std::cout << "histogram matching: cropping target to get an aspect ratio of " << std::fixed << std::setprecision(2) << thumb_ratio << ":1, new full size is " << fw << "x" << fh << std::endl;
+                std::cout << "histogram matching: cropping target to get an aspect ratio of " << std::fixed << std::setprecision(2) << thumb_ratio << ":1, new size is " << tw << "x" << th << std::endl;
```

Comment:
That's the drawback of streams: They aren't stateless. Coverity already warned about `std::cout` being changed without reseting...